Automattic/newspack-popups
# Handle multiple popups

### All Submissions:






### Changes proposed in this Pull Request:



A couple of changes to the logic and architecture, which result in the possibility of having multiple inline popups, and at most one overlaid popup:

- In editor, the "Sitewide Default" switch will be disabled if "inline" is chosen as placement.
- Some checks on the backend will prevent an inline popup from being marked as "sitewide default".
- Inline popups are inserted as shortcodes. This prevents an inline popup ending up inside another inline popup.
- All inline popups will be inserted into a post, unless there's category filtering. Any popup with a set category will only appear on a post with matching category.
- A non-inline popup will be displayed only if it's sitewide default, as it were (unless there's category filtering)
-  In admin's popup list view, a post state of "Inline" is added next to inline popups.

Closes #77 

### How to test the changes in this Pull Request:

1. Create a lot of popups - both inline and overlaid. Go wild.
2. Browse the site - all inline popups should be visible on posts.
3. Mark one of overlay (non-inline) popups as "Sitewide default".
4. Browse the site - apart from the inline popups, the overlaid sitewide popup should be displayed.

### Other information:
